### PR TITLE
feat: upgrade openetelemetry to 0.18

### LINKED ISSUE #235: Add support for 0.17 opentelemetry-js package
*Is your feature request related to a problem? Please describe.*
A new version of the `opentelemetry`-package has been release version v0.17

*Describe the solution you'd like*
Update the package to depend on v0.17 instead of v0.16

*Additional context*
None

### PR DESCRIPTION
**Pull request by tapico-weyert**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Make sure to open an issue as a [bug/issue/feature](https://github.com/GoogleCloudPlatform/opentelemetry-operations-js/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #235, #236

### CODE REVIEW

**Comment by tapico-weyert** [comment]
@googlebot I signed it!

**Comment by tapico-weyert** [comment]
@googlebot I signed it!

File: packages/opentelemetry-cloud-monitoring-exporter/test/monitoring.test.ts [hunk 1]
```diff
@@ -133,10 +125,10 @@ describe('MetricExporter', () => {
...
       warn = sinon.spy();
       error = sinon.spy();
-      sinon.replace(logger, 'debug', debug);
-      sinon.replace(logger, 'info', info);
-      sinon.replace(logger, 'warn', warn);
-      sinon.replace(logger, 'error', error);
+      sinon.replace(diag, 'debug', debug);
+      sinon.replace(diag, 'info', info);
+      sinon.replace(diag, 'warn', warn);
+      sinon.replace(diag, 'error', error);
```
Comment (aabmass) [inline packages/opentelemetry-cloud-monitoring-exporter/test/monitoring.test.ts:131]: Looks like these weren't used at all, can you delete these variables?

File: packages/opentelemetry-cloud-trace-exporter/test/exporter.test.ts [hunk 2]
```diff
@@ -174,15 +165,14 @@ describe('Google Cloud Trace Exporter', () => {
...
       warn = sinon.spy();
       error = sinon.spy();
-      sinon.replace(logger, 'debug', debug);
-      sinon.replace(logger, 'info', info);
-      sinon.replace(logger, 'warn', warn);
-      sinon.replace(logger, 'error', error);
+      sinon.replace(diag, 'debug', debug);
+      sinon.replace(diag, 'info', info);
+      sinon.replace(diag, 'warn', warn);
+      sinon.replace(diag, 'error', error);
```
Comment (aabmass) [inline packages/opentelemetry-cloud-trace-exporter/test/exporter.test.ts:175]: Same as above, can we delete these variables altogether?

File: packages/opentelemetry-cloud-trace-propagator/package.json [hunk 3]
```diff
@@ -38,8 +38,8 @@
     "registry": "https://wombat-dressing-room.appspot.com"
   },
   "devDependencies": {
-    "@opentelemetry/api": "^0.16.0",
-    "@opentelemetry/core": "^0.16.0",
+    "@opentelemetry/api": "^0.18.0",
+    "@opentelemetry/core": "^0.17.0",
```
Comment (aabmass) [inline packages/opentelemetry-cloud-trace-propagator/package.json:42]: Same as above

File: packages/opentelemetry-cloud-trace-exporter/package.json [hunk 4]
```diff
@@ -38,9 +38,9 @@
     "registry": "https://wombat-dressing-room.appspot.com"
   },
   "devDependencies": {
-    "@opentelemetry/api": "^0.16.0",
-    "@opentelemetry/resources": "^0.16.0",
-    "@opentelemetry/tracing": "^0.16.0",
+    "@opentelemetry/api": "^0.18.0",
+    "@opentelemetry/resources": "^0.17.0",
+    "@opentelemetry/tracing": "^0.17.0",
```
Comment (aabmass) [inline packages/opentelemetry-cloud-trace-exporter/package.json:43]: Same as above

**Review by aabmass — CHANGES_REQUESTED** [review]
Just a few quick change requests, but otherwise LGTM. Can you also use npm v7 to run `npm install` so we get `lockfileVersion` 2? If you have trouble with it, I can add to the PR 😃

File: packages/opentelemetry-cloud-monitoring-exporter/package.json [hunk 5]
```diff
@@ -62,9 +62,9 @@
...
   },
   "peerDependencies": {
-    "@opentelemetry/api": "^0.16.0",
-    "@opentelemetry/core": "^0.16.0",
-    "@opentelemetry/metrics": "^0.16.0",
-    "@opentelemetry/resources": "^0.16.0"
+    "@opentelemetry/api": "^0.18.0",
+    "@opentelemetry/core": "^0.17.0",
+    "@opentelemetry/metrics": "^0.17.0",
+    "@opentelemetry/resources": "^0.17.0"
```
[the remark this answers is not in the document]
Comment (weyert) [reply packages/opentelemetry-cloud-monitoring-exporter/package.json:68]: I will update them

File: packages/opentelemetry-cloud-trace-exporter/test/exporter.test.ts (hunk 2 above)
Comment (weyert) [reply packages/opentelemetry-cloud-trace-exporter/test/exporter.test.ts:175]: The `error` is one is being used on line 290 I have left that one in but removed the others

**Review by aabmass — APPROVED** [review]
🥇 thank you again for the PR @tapico-weyert

**Pull request merged**